src/modules/hmis/components/ClientSsn.tsx

```diff
@@ -26,4 +28,28 @@ const ClientSsn = ({ client, noValue, lastFour, variant = 'body2' }: Props) => {
   );
 };
 
+export const ClientSafeSsn: React.FC<Props> = (props) => {
+  const { client } = props;
+
+  const [canViewFullSsn] = useHasClientPermissions(client.id, [
+    'canViewFullSsn',
+  ]);
+  const [canViewPartialSsn] = useHasClientPermissions(client.id, [
+    'canViewPartialSsn',
+  ]);
```

Comment:
Can we skip these queries if `client.ssn` is already null? The backend won't return the SSN at all if it can't be seen, so it's unnecessary to do an extra check.

Reply:
@gigxz Yes, we can skip the `canViewFullSsn` query if it's null. However, the `canViewPartialSsn` is not as straightforward. If the SSN is `123456789`, the backend will return `XXXXX6789`, but that is also a valid full SSN, so without checking the permission we can't know whether it only gave us the last four because that's all that we have for that client, or whether it's because we only have permission to see the partial SSN.

If you're okay with it we could just add the permissions to the `ClientIdentificationFieldsFragment` so we just fetch it with the rest of the fields and then should have it present within the SSN component. That feels like the easiest way to do it. Does that sound good to you?

Reply:
Hey @ZachCMP sorry for the delayed response on this one. I went ahead and made that change while reviewing so you don't need to context switch back to this: https://github.com/greenriver/hmis-frontend/pull/175/commits/c017a3408afb4f29952af0bcb1811aaad7c83d3b

Overall it's looking great, there is just a performance issue: https://github.com/greenriver/hmis-warehouse/pull/2868#discussion_r1135459887